client/src/components/UserProfile.jsx

```diff
@@ -14,6 +14,12 @@ import { Spinner } from 'react-bootstrap';
 
 class _UserProfile extends React.Component {
 
+  componentDidMount() {
```

Comment:
Couple of things here:
1.) It may not show up as an issue in testing locally, but there's a race condition here with the `users/check` in App.jsx and this component mounting. That's why the call to `this.props.getUser()` is called after the user state is validated. It's possible that `users/check` may not resolve when the calls to `users/:id/reviews` and `users/:id/collections`  goes out in `componentDidMount()`  since the user page is dependent on the user state being checked in App.jsx, there's not a need to load any external data with the component mounts. 
2.) The call to `getUser()` already handles both user reviews and collections. It should probably be renamed though since that isn't very clear.